### PR TITLE
feat: ARO-11963 | add outbound type changes in OCM SDK Go

### PR DESCRIPTION
**Pull request by ziccardi**

### CODE REVIEW

File: model/clusters_mgmt/v1/azure_type.model [hunk 1]
```diff
@@ -101,4 +101,8 @@ struct Azure {
 	// Required during creation.
 	// Immutable.
 	OperatorsAuthentication AzureOperatorsAuthentication
+
+	// NodesOutboundConnectivity defines how the network outbound
```
Comment (miguelsorianod) [inline model/clusters_mgmt/v1/azure_type.model:105]: There's some whitespacing issue here

File: model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The configuration of the node outbound connectivity
+struct AzureNodesOutboundConnectivity {
+	// OutboundType is the type of network outbound configuration.
+	// The only accepted value is 'load_balancer'
```
Comment (miguelsorianod) [inline model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model:20]: We should document that the default is `load_balancer`. Yes, `load_balancer` is the only accepted value (for now), but it should be clear that a default is set for it, the user doesn't explicitly need to set it. I'd also document that it is Immutable

File: model/clusters_mgmt/v1/azure_type.model [hunk 3]
```diff
@@ -101,4 +101,8 @@ struct Azure {
...
+	// configuration of the Cluster's Node Pool's Nodes is performed.
+	NodesOutboundConnectivity AzureNodesOutboundConnectivity
```
Comment (miguelsorianod) [inline model/clusters_mgmt/v1/azure_type.model:107]: Here we should document that by default the network outbound configuration is through an Azure Load Balancer, and that it is immutable.

File: model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+limitations under the License.
+*/
+
+// The configuration of the node outbound connectivity
+struct AzureNodesOutboundConnectivity {
+	// OutboundType is the type of network outbound configuration.
+	// The default and only accepted value is 'load_balancer'.
+	// This value is immutable.
+	OutboundType String
+}
```
Comment (machi1990) [inline model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model:23]:
```suggestion
}

```
nit; missing newline

**Review by machi1990 — APPROVED** [review]
left one nit, but lgtm

File: model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model (hunk 4 above)
Comment (ziccardi) [reply model/clusters_mgmt/v1/azure_nodes_outbound_connectivity_types.model]: fixed!

**Review by machi1990 — APPROVED** [review]

**Pull request merged**